src/commands/markov.rs

```diff
@@ -54,7 +57,61 @@ command!(generate_user(ctx, message, args) {
         if let Some(markov) = get_uid(arg)
                 .map_or(None,
                         |x| usermap.get(&x)) {
-            output_markov(&markov, &message, length);
+            output_markov(&markov, &message, length, None);
+        }
+    }
+});
+
+
+command!(generate_from_word(ctx, message, args) {
+    let mut data = ctx.data.lock().unwrap();
+    let mut markov = data.get_mut::<Markov>().unwrap();
+    let mut word: Option<&str> = None;
+
+    let mut length: u32 = args.get(1)
+        .map_or(DEFAULT_GENERATION_LENGTH,
+                |x| x.parse::<u32>()
+                .ok()
+                .unwrap_or(DEFAULT_GENERATION_LENGTH));
+    
+    match args.get(0) {
+        Some(arg) => {
+            word = Some(&arg.as_str());
+        }
+
+        None => {
+            let _ = message.channel_id.say("This command needs to take a word as the first argument");
         }
     }
+    output_markov(&markov, &message, length, word);
+});
+
+
+command!(generate_word_from_user(ctx, message, args) {
+    let mut data = ctx.data.lock().unwrap();
+    let mut usermap = data.get_mut::<UserMap>().unwrap();
+    let mut word: Option<&str> = None;
+        
+    let mut length: u32 = args.get(2)
+        .map_or(DEFAULT_GENERATION_LENGTH,
+                |x| x.parse::<u32>()
+                .ok()
+                .unwrap_or(DEFAULT_GENERATION_LENGTH));
+
+    if let Some(arg) = args.get(0) {
+        if let Some(markov) = get_uid(arg)
+            .map_or(None,
+                    |x| usermap.get(&x)) {
+                word = args.get(1).map_or(None,
+                                          |x| Some(x.as_str()));
+            }
+    }
+        
+    if let Some(arg) = args.get(0) {
+        if let Some(markov) = get_uid(arg)
+            .map_or(None,
+                    |x| usermap.get(&x)) {
+                output_markov(&markov, &message, length, word);
+            }
+    }
```

Comment:
This looks basically identical to your previous command.

Why can't you refactor into a helper function that both channel into?